lib/catalog/owid/catalog/catalogs.py

```diff
@@ -108,7 +133,12 @@ def find(
             criteria &= self.frame.version == version
 
         if dataset:
-            criteria &= self.frame.dataset == dataset
+            if fuzzy:
+                scores = self._fuzzy_scores(self.frame.dataset, dataset, case)
+                criteria &= scores >= threshold
+                fuzzy_scores = scores if fuzzy_scores is None else fuzzy_scores + scores
+            else:
+                criteria &= self.frame.dataset.str.contains(dataset, case=case, regex=regex)
```

Comment:
I'm realising this block can be abstracted and re-used

```python
def get_criteria_and_fuzzy_scores(self, object_type: str, obj_value, criteria, fuzzy_scores, threshold, case, regex):
  if fuzzy:
    scores = self._fuzzy_scores(self.frame[object_type], obj_value, case)
    criteria &= scores >= threshold
    fuzzy_scores = scores if fuzzy_scores is None else fuzzy_scores + scores
  else:
    criteria &= self.frame[object_type].str.contains(obj_value, case=case, regex=regex)
```

i'm also realising how many args this function has, so not super nice. But I guess better than duplicating code?